org.wso2.codequalitymatrices/src/main/java/com/wso2/code/quality/matrices/ApiUtility.java

```diff
@@ -0,0 +1,170 @@
+            httpResponse = httpClient.execute(httpGet);
+            int responseCode = httpResponse.getStatusLine().getStatusCode();
+            if (responseCode == 200) {
+                //success
+                bufferedReader = new BufferedReader(new InputStreamReader(httpResponse.getEntity().getContent(),
+                        "UTF-8"));
+                StringBuilder stringBuilder = new StringBuilder();
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
+                    stringBuilder.append(line);
+                }
+                // creating a JSON object from the response
+                jsonText = stringBuilder.toString();
+            } else {
+                throw new CodeQualityMatricesException("Error occurred while calling the API, the response code is " +
+                        responseCode);
+            }
+        } catch (ClientProtocolException e) {
+            throw new CodeQualityMatricesException("ClientProtocolException when calling the REST API", e);
+        } catch (IOException e) {
+            throw new CodeQualityMatricesException("IOException occurred when calling the REST API", e);
+        } finally {
+            if (bufferedReader != null) {
+                try {
+                    bufferedReader.close();
+                } catch (IOException e) {
+                    logger.debug("IOException occurred when closing the BufferedReader", e);
+                }
+            }
+            if (httpResponse != null) {
+                try {
+                    httpResponse.close();
+                } catch (IOException e) {
+                    logger.debug("IOException occurred when closing the HttpResponse", e);
+                }
+            }
+            if (httpClient != null) {
+                try {
+                    httpClient.close();
+                } catch (IOException e) {
+                    logger.debug("IOException occurred when closing the HttpClient", e);
+                }
+            }
+        }
+        return jsonText;
+    }
+
+    /**
+     * This is used for calling the github graphql API.
+     *
+     * @param httpPost relevant instance of the httpost
+     * @return String representation of the json response
+     * @throws CodeQualityMatricesException
+     */
+    public static String callGraphQlApi(HttpPost httpPost) throws CodeQualityMatricesException {
+        BufferedReader bufferedReader = null;
+        CloseableHttpClient httpClient;
+        CloseableHttpResponse httpResponse = null;
+        httpClient = HttpClients.createDefault();
+        String jsonText;
+        try {
+            httpResponse = httpClient.execute(httpPost);
+            int responseCode = httpResponse.getStatusLine().getStatusCode();
+            if (responseCode == 200) {
+                bufferedReader = new BufferedReader(new InputStreamReader(httpResponse.getEntity().getContent(),
+                        "UTF-8"));
```

Comment:
Here also please use the given constants as nisala mentioned above instead of using string.